nuxeo-runtime/nuxeo-connect-standalone/src/test/java/org/nuxeo/launcher/connect/TestConnectBroker.java

```diff
@@ -179,6 +192,62 @@ public void afterEach() {
         System.clearProperty(TomcatConfigurator.TOMCAT_HOME);
     }
 
+    // NXP-24507
+    @Test
+    public void testInstallPackageRequest_restartLauncherWithNoPending() {
+        // Given a downloaded package, which requires launcher restart
+        checkPackagesState(connectBroker, Arrays.asList("NXP-24507-A-1.0.0"), PackageState.DOWNLOADED);
+
+        // When handling the install request
+        try {
+            connectBroker.pkgRequest(null, Arrays.asList("NXP-24507-A-1.0.0"), null, null, true, false);
+            fail();
+        } catch (LauncherRestartException e) {
+            // Then an exception is raised
+        }
```

Comment:
would be better to do something like `assertTrue(connectBroker.isRestartRequired(...))`...